lib/haml_lint/reporter/json_reporter.rb

```diff
@@ -0,0 +1,34 @@
+module HamlLint
+  # Outputs JSON
+  class Reporter::JsonReporter < Reporter
+    def report_lints
+      grouped = lints.group_by(&:filename)
+      report = {
+        :metadata => {
+          :hamllint_version => VERSION,
+          :ruby_engine      => RUBY_ENGINE,
+          :ruby_patchlevel  => RUBY_PATCHLEVEL.to_s,
+          :ruby_platform    => RUBY_PLATFORM
+        },
+        :files    => grouped.map { |l| map_file(l) },
+        :summary  => {
+          :offense_count        => lints.length,
+          :target_file_count    => grouped.length,
+          :inspected_file_count => files.length
```

Comment:
@sds, sccording to the [RuboCop specification](http://www.rubydoc.info/gems/rubocop/0.11.0#JSON_Formatter), the summary section of the JSON output has the number of the inspected files. I didn't wanted to pass just the number, because this way it's possible to create e.g. a "positive-reporter", which shows you the files without any offenses.